cmake/FindONNX.cmake

```diff
@@ -0,0 +1,31 @@
+ocv_clear_vars(HAVE_ONNX)
+
+set(ORT_INSTALL_DIR "" CACHE PATH "ONNX Runtime install directory")
+
+if(ORT_INSTALL_DIR)
+  find_library(ORT_LIB onnxruntime
+    ${ORT_INSTALL_DIR}/lib
+    CMAKE_FIND_ROOT_PATH_BOTH)
+  find_path(ORT_INCLUDE onnxruntime_cxx_api.h
+    ${ORT_INSTALL_DIR}/include/onnxruntime/core/session
+    CMAKE_FIND_ROOT_PATH_BOTH)
+endif()
+
+if(ORT_LIB AND ORT_INCLUDE)
+  set(HAVE_ONNX TRUE)
+  # For CMake output only
+  set(ONNX_LIBRARIES "${ORT_LIB}" CACHE STRING "ONNX Runtime libraries")
+  set(ONNX_INCLUDE_DIR "${ORT_INCLUDE}" CACHE STRING "ONNX Runtime include path")
+
+  # Link target with associated interface headers
+  set(ONNX_LIBRARY "ort" CACHE STRING "ONNX Link Target")
+  ocv_add_library(${ONNX_LIBRARY} SHARED IMPORTED)
+  set_target_properties(${ONNX_LIBRARY} PROPERTIES
+                        INTERFACE_INCLUDE_DIRECTORIES ${ORT_INCLUDE}
+                        IMPORTED_LOCATION ${ORT_LIB}
+                        IMPORTED_IMPLIB ${ORT_LIB})
```

Comment:
after that

```
set(ONNX_LIBRARIES "${ONNX_LIBRARY}")
```

should be applied.

Reply:
Then CMake output will text about our internal name of this library, but won't show its full actual path